### PR TITLE
[#19410] wizard send collectibles

### LINKED ISSUE #16990: Implement Send collectible from Collectibles details page
# Feature Issue

### User Story

As a user, I want to be able to send a collectible from the collectible detail screen so that I can quickly initiate the sending process without navigating back to the account page, streamlining the overall experience.

### Description

Pre-requisites: Collectible details screen should be implemented.

This issue is more related to flows rather than implementing UI, UI should be already implemented at this point.

Figma design 1: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1355-315956&mode=design&t=PQXlpA0Kw075Q09Y-4
Figma design 2: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1355-316125&mode=design&t=PQXlpA0Kw075Q09Y-4
Figma design 3: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1355-317327&mode=design&t=PQXlpA0Kw075Q09Y-4

Visual reference:
 

### Acceptance Criteria

- [ ] When user taps on `Send` button within a Collectible detail screen, should navigate to Select Address page
- [ ] When user inputs or selects an Address / Contact / Account / Saved Address, should navigate to Transaction Confirmation page to confirm the transaction (#16979)

### LINKED ISSUE #19410: Wallet - Skip steps while sending an asset picked from an account overview
# Bug Report

## Problem 
[comment]: # (A problem description)

We need to use the recently created wizard mechanism to skip the needed steps while sending an asset from the account overview screen in the wallet.

By long pressing an asset, we get the following image:

The flow to fix is the **Send** one.

Figma:
https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=7701-154896&mode=design&t=VEZquM5jvvYIUGvQ-4

This issue is also present in other flows:

- Send a Collectible: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1355-315956&mode=design&t=VEZquM5jvvYIUGvQ-4

Split this issue if needed

#### Expected behavior

Skip steps that add already known information

#### Actual behavior

All the steps are presented to the user

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ulisesmac**
fixes #19410 
fixes #16990

The bridge feature will be addressed in
- #19516

### Summary

This PR uses the wizard mechanism to unify the send token and send collectible flows, making able to skip some steps for the sending of collectibles.

Video containing the regular flow and the flow of first picking a collectible (skips select asset screen):

[Screencast from 2024-04-03 13-13-04.webm](https://github.com/status-im/status-mobile/assets/90291778/4927e2ab-f7f7-4b1a-93cc-9b010feab71a)

Flow from home wallet screen (adds from step and skips sleect asset):

[Screencast from 2024-04-03 13-14-38.webm](https://github.com/status-im/status-mobile/assets/90291778/9c35b401-db7f-4f52-b814-6538e1df78b9)

Additionally, solves the following error when going back in the navigation:

[Screencast from 2024-04-03 13-02-27.webm](https://github.com/status-im/status-mobile/assets/90291778/1fa055d2-4185-4841-ac23-b9fabbe66bbb)

### Testing notes
This PR focuses on the flow, there may be some other areas that still need to be polished/adjusted.

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Use/Recover an account owning some collectibles
- Go to the account details
- Move to Collectibles tab
- Press any collectible
- Press "Send" and complete the send flow.

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/collectible/view.cljs [hunk 1]
```diff
@@ -96,11 +101,11 @@
...
         :right-icon          :i/external}]]]))
 
-(defn f-view-internal
-  [{:keys [theme] :as _props}]
+(defn view-internal
+  [_]
   (let [selected-tab  (reagent/atom :overview)
         on-tab-change #(reset! selected-tab %)]
-    (fn []
+    (fn [{:keys [theme] :as _props}]
```
Comment (ulisesmac) [inline src/status_im/contexts/wallet/collectible/view.cljs]: An additional fix, we were receiving the `theme` in the wrong place, the component wouldn't re-render if it changed

File: src/status_im/contexts/wallet/send/events.cljs [hunk 2]
```diff
@@ -99,27 +100,37 @@
...
-    :fx [[:navigate-to-within-stack [:screen/wallet.select-collectible-amount stack-id]]]}))
+(rf/reg-event-fx
+ :wallet/set-collectible-amount-to-send
+ (fn [{db :db} [{:keys [stack-id amount]}]]
+   {:db (assoc-in db [:wallet :ui :send :amount] amount)
+    :fx [[:dispatch [:wallet/get-suggested-routes {:amount amount}]]
+         [:dispatch
+          [:wallet/wizard-navigate-forward
+           {:current-screen stack-id
+            :flow-id        :wallet-flow}]]]}))
```
Comment (ulisesmac) [inline src/status_im/contexts/wallet/send/events.cljs]: Redefined these events to handle the wizard flow, and changed their names.

`send-collectibles-amount` -> `set-collectible-to-send`
`select-collectibles-amount` -> `set-collectible-amount-to-send`

Given the functionality changed and the places where they are being used, the new names match better

File: src/status_im/contexts/wallet/send/events.cljs [hunk 3]
```diff
@@ -99,27 +100,37 @@
...
 
-(rf/reg-event-fx :wallet/send-select-amount
+(rf/reg-event-fx
+ :wallet/set-token-amount-to-send
```
Comment (ulisesmac) [inline src/status_im/contexts/wallet/send/events.cljs:255]: According to the previous mentioned renaming, and  given that this even is specific for tokens and not collectibles, I performed this rename

File: src/status_im/contexts/wallet/send/select_asset/view.cljs [hunk 4]
```diff
@@ -94,8 +87,4 @@
          :on-change       on-change-tab}]
        [tab-view @search-text @selected-tab on-change-text]])))
 
-(defn- view-internal
-  []
-  [:f> f-view-internal])
```
Comment (ulisesmac) [inline src/status_im/contexts/wallet/send/select_asset/view.cljs]: Just removed this (now unnecessary) wrapper

File: src/status_im/contexts/wallet/send/events.cljs [hunk 5]
```diff
@@ -99,27 +100,37 @@
                      dissoc
                      :collectible
                      :amount
-                     (when (= transaction-type :collecible) :tx-type))})))
+                     (when (= transaction-type :collectible) :tx-type))})))
```
Comment (ulisesmac) [inline src/status_im/contexts/wallet/send/events.cljs:213]: This fix is not directly related, but noticed we had a typo here :eyes:

File: src/status_im/contexts/wallet/send/events.cljs [hunk 6]
```diff
@@ -99,27 +100,37 @@
...
 
-(rf/reg-event-fx :wallet/select-collectibles-amount
- (fn [{:keys [db]} [{:keys [collectible stack-id]}]]
-   {:db (-> db
-            (update-in [:wallet :ui :send] dissoc :token)
-            (assoc-in [:wallet :ui :send :collectible] collectible)
-            (assoc-in [:wallet :ui :send :tx-type] :collectible))
-    :fx [[:navigate-to-within-stack [:screen/wallet.select-collectible-amount stack-id]]]}))
+(rf/reg-event-fx
+ :wallet/set-collectible-amount-to-send
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/send/events.cljs]: I think we can have tests for these events

**Comment by J-Son89** [comment]
@ulisesmac - I tested out this branch. 
It works great for sending an asset.
One thought - there is also sending an asset from the home page by long press on a collectible.
@mmilad75 has added the "Select From" page for this flow already so it should be rather straightforward to include that. 
Shall we add it here too? 🤔

File: src/status_im/contexts/wallet/send/flow_config.cljs [hunk 7]
```diff
@@ -1,13 +1,30 @@
...
+  (let [collectible-stored (-> db :wallet :ui :send :collectible)
+        tx-type            (-> db :wallet :ui :send :tx-type)]
+    (and (some? collectible-stored)
+         (= tx-type :collectible))))
+
+(defn- token-selected?
+  [db]
+  (-> db :wallet :ui :send :token some?))
+
 (def steps
```
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/flow_config.cljs:14]: @ulisesmac - can you instead make a separate flow for sending an asset vs a collectible.
I think it will be cleaner this way. We should also do the same for bridging imo. 

There's also an issue with sending a collectible this way. 

The issue is this, if you are on the home page - long press on a collectible.
When you click send it will bring you to the "From" Page.
This From page shows all addresses 
- it's unclear if we want this, probably we want to show that page only if that collectible is owned by many accounts by the user.


Also in the case of a user do we really want to allow a user to select an account that doesn't have that asset to send? 🤔 
Seems like the other accounts should be greyed out, probably a separate issue.

File: src/status_im/contexts/wallet/send/events.cljs [hunk 8]
```diff
@@ -99,27 +100,37 @@
...
+                              (assoc-in [:wallet :ui :send :collectible] collectible)
+                              (assoc-in [:wallet :ui :send :tx-type] :collectible))]
+     {:db (cond-> collectible-tx
+            one-collectible? (assoc-in [:wallet :ui :send :amount] 1))
+      :fx [(when one-collectible?
+             [:dispatch [:wallet/get-suggested-routes {:amount 1}]])
+           [:dispatch
+            [:wallet/wizard-navigate-forward
+             {:current-screen current-screen
+              :flow-id        :wallet-flow}]]]})))
```
Comment (J-Son89) [inline src/status_im/contexts/wallet/send/events.cljs]: the idea here was to have the different flows marked.
i.e :asset-flow, :bridge-flow, :collectible-flow 👍 

I think we should use it this way and not mix the logic of these as they are all slightly different and it will be cleaner 👍

File: src/status_im/contexts/wallet/send/flow_config.cljs (hunk 7 above)
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/flow_config.cljs:14]: Yeah, since while sending a token, we also display all accounts, I wasn't sure if we really wanted to filter.

Maybe better if this filtering for sending collectibles and assets is first calrified and then solved in the different wallet areas?

Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/flow_config.cljs:14]: Regarding separating the flow, at the beginning it was my approach, but this infrastructure doesn't support it.

At the beginning, I was in a flow (the regular sending) and then just jumped to another flow when a collectible was picked (`wallet-flow` -> `send-collectible-flow`) and it worked for the simple sending, but once I wanted to skip some steps (particularly the collectible picking) then the flow didn't know it must jump to another flow.

So I realized that if I wanted that same screens take different flows and skip steps successfully, I should, from the beginning, stablish a flow and persist it on re-frame, and not passing the flow-id to the wizard anymore, UNLESS, you wanted to jump from one flow to another, and (IDK exactly how and when) make sure to clean the db with the flow picked, because the user could just give up, and close the flow even if it's not completed.


Since this required a significant change in the structure I thought it was not intended to do so, so I just kept our `wallet-flow` sharing all steps but skipping according to conditions based on the type of transaction.

I'm not against making big changes to the mechanism, I just was unsure, so if you think that is the best, I can do it 👍

Also, idk if we want multiple flows repeating many steps (as firsts and last steps in sending) or we want to define a flow with subflows (maybe a bit more complex to implement).

Wdyt?

File: src/status_im/contexts/wallet/send/events.cljs (hunk 8 above)
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/events.cljs]: It was due to the infra not supporting it. Answered in https://github.com/status-im/status-mobile/pull/19515#discussion_r1551949724

**Comment by ulisesmac** [comment]
> @ulisesmac - I tested out this branch. 
> It works great for sending an asset.
> One thought - there is also sending an asset from the home page by long press on a collectible.
> @mmilad75 has added the "Select From" page for this flow already so it should be rather straightforward to include that. 
> Shall we add it here too? 🤔 

Sure!, I'll add it :)

**Comment by ulisesmac** [comment]
> > @ulisesmac - I tested out this branch.
> > It works great for sending an asset.
> > One thought - there is also sending an asset from the home page by long press on a collectible.
> > @mmilad75 has added the "Select From" page for this flow already so it should be rather straightforward to include that.
> > Shall we add it here too? 🤔
> 
> Sure!, I'll add it :)

I already added it, it wasn't as straightforward as I thought but it's not too complex, I added two comments in the code to make this more clear.

**Comment by ulisesmac** [comment]
@OmarBasem @J-Son89 
Can you please review again?
additionally, I think QA can be skipped, wdyt?

**Comment by J-Son89** [comment]
> @OmarBasem @J-Son89 Can you please review again? additionally, I think QA can be skipped, wdyt?

I tend to disagree, I think this will benefit from QA. Send is quite an important area and we're making quite a number of changes which can break it, no?

**Comment by ulisesmac** [comment]
> > @OmarBasem @J-Son89 Can you please review again? additionally, I think QA can be skipped, wdyt?
> 
> I tend to disagree, I think this will benefit from QA. Send is quite an important area and we're making quite a number of changes which can break it, no?

I thought this was kind of just connecting existing features, but I agree, let's ask for QA then :+1:

File: src/status_im/contexts/wallet/send/flow_config.cljs [hunk 9]
```diff
@@ -1,13 +1,30 @@
...
+    :skip-step? (fn [db] (or (token-selected? db) (collectible-selected? db)))}
    {:screen-id  :screen/wallet.send-input-amount
-    :skip-step? (fn [db] (some? (get-in db [:wallet :ui :send :amount])))}
+    :skip-step? (fn [db]
+                  (or (not (token-selected? db))
+                      (some? (get-in db [:wallet :ui :send :amount]))))}
+   {:screen-id  :screen/wallet.select-collectible-amount
+    :skip-step? (fn [db]
+                  (or (not (collectible-selected? db))
+                      (some? (get-in db [:wallet :ui :send :amount]))))}
```
Comment (briansztamfater) [inline src/status_im/contexts/wallet/send/flow_config.cljs:28]: `amount` should be greater than 1, so probably we should skip if amount == 1 because user could only send 1  🤔

Comment (J-Son89) [reply src/status_im/contexts/wallet/send/flow_config.cljs:28]: we have multiple collectibles now :)

**Comment by J-Son89** [comment]
> > > @OmarBasem @J-Son89 Can you please review again? additionally, I think QA can be skipped, wdyt?
> > 
> > 
> > I tend to disagree, I think this will benefit from QA. Send is quite an important area and we're making quite a number of changes which can break it, no?
> 
> I thought this was kind of just connecting existing features, but I agree, let's ask for QA then 👍

kind of, but the extent to which QA scrutinise this can be shallower. Please make sure to note that in the testing steps 👍

File: src/status_im/contexts/wallet/collectible/view.cljs [hunk 10]
```diff
@@ -28,13 +29,17 @@
...
-  []
+  [collectible]
   [rn/view {:style style/buttons-container}
    [quo/button
     {:container-style style/send-button
      :type            :outline
      :size            40
-     :icon-left       :i/send}
+     :icon-left       :i/send
+     :on-press        #(rf/dispatch
```
Comment (J-Son89) [inline src/status_im/contexts/wallet/collectible/view.cljs]: 🙌 
(also this is one reason it's good to test. It's technically new functionality 👍

File: src/status_im/contexts/wallet/collectible/view.cljs (hunk 1 above)
Comment (J-Son89) [reply src/status_im/contexts/wallet/collectible/view.cljs]: can also use the hook directly, if you prefer 👍

File: src/status_im/contexts/wallet/send/flow_config.cljs (hunk 9 above)
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/flow_config.cljs:28]: this `amount` is not the owned collectible units, this `amont` key represents the desired number of collectibles to send and we are skipping this step if it already exists in the db.
i.e.
During this step we set the value in `[:wallet :ui :send :amount]`, if it has been already set, we skip it.

**Review by J-Son89 — APPROVED** [review]

File: src/status_im/contexts/wallet/send/flow_config.cljs (hunk 9 above)
Comment (J-Son89) [reply src/status_im/contexts/wallet/send/flow_config.cljs:28]: yes, but the user can decide how many at a time they want to send. 
see @vkjr's recent pr:
https://github.com/status-im/status-mobile/pull/19386

Comment (J-Son89) [reply src/status_im/contexts/wallet/send/flow_config.cljs:28]: i,e https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=10771-86609&mode=design&t=XuAtw4Ji54tLkXJS-4

although, maybe we're talking about the same thing? 
🤔 
in any case this should only be for collectibles that are owned, so if this is a stretch in functionality, let's do a follow up instead.

**Review by OmarBasem — APPROVED** [review]
Looks good, but would be better to add tests for the new/updated send events as it is a sensitive functionality and I think it broke a few times before

File: src/status_im/contexts/wallet/send/flow_config.cljs (hunk 9 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/flow_config.cljs:28]: > this `amount` is not the owned collectible units, this `amont` key represents the desired number of collectibles to send and we are skipping this step if it already exists in the db. i.e. During this step we set the value in `[:wallet :ui :send :amount]`, if it has been already set, we skip it.

I see, thanks for the explanation @ulisesmac.

> yes, but the user can decide how many at a time they want to send. see @vkjr's recent pr: #19386

@J-Son89 What I meant is that if user only owns 1 of the selected collectible, it doesn't make sense to show `:screen/wallet.select-collectible-amount`. But I see the case is already covered by this part
```
     {:db (cond-> collectible-tx
            one-collectible? (assoc-in [:wallet :ui :send :amount] 1))
```
in `:wallet/set-collectible-to-send` event, and the condition in skip-step?.

Looks good to me 👍

**Review by briansztamfater — APPROVED** [review]

**Comment by briansztamfater** [comment]
BTW I agree we would benefit of QA review as send flows are a sensitive part and core functionality of the wallet 👍

File: src/status_im/contexts/wallet/send/flow_config.cljs (hunk 9 above)
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/flow_config.cljs:28]: @J-Son89 
Yeah, that flow is being preserved, and it's only shown when the user has more than one collectible.

ok! @briansztamfater  :)

**Comment by ulisesmac** [comment]
> Looks good, but would be better to add tests for the new/updated send events as it is a sensitive functionality and I think it broke a few times before

I considered it before, but I also thought these events modified are just adding like 1 or 2 keys, they mainly rely on the correctness of the wallet flow. But yeah, I can add some test just to check those keys exist.

**Review by vkjr — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @ulisesmac . Could you please resolve existing conflicts and rebase PR? thanx!

**Comment by ulisesmac** [comment]
> Hi @ulisesmac . Could you please resolve existing conflicts and rebase PR? thanx!

@VolodLytvynenko Done!

**Comment by VolodLytvynenko** [comment]
Hi @ulisesmac Thank you for PR. Take a look at the found issue:

## PR_ISSUE 1: The assets can't be sent on 'send to' page when an asset is selected on account overview page

### Steps:
1. Restore the user with available assets (example: eth)
2. Go to the account overview page
3. Long tap on the asset -> send
4. Select account with available assets to send from (eth)
5. Navigate to sent to page

### Actual result:
User sees 0 assets on the 'send to' page.

https://github.com/status-im/status-mobile/assets/52490791/27f047b6-a356-4fcc-9638-925f31cbe5ff

### Expected result:
User sees available assets on the 'sent to' page and can sent assets

### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by VolodLytvynenko** [comment]
I'm not sure if this issue is within the scope of this PR. I think it would be better to implement the collectible detail page for watch-only collectibles before this PR. This is because currently, this PR allows going to the send flow for watch-only collectibles. 

However, this could be considered as a follow-up. Is it possible to implement this page within the current PR, or would it be better to do it separately? WDYT?

## PR_ISSUE 2: 'Send' button is shown for watch-only collectibles and can navigate to the 'send' flow

### Step:
1. Add watch only account with some collectibles (example: `0x8d2413447ff297d30bdc475f6d5cb00254685aae`. Take a look some collectibles are not fetched until relogin due existing issue https://github.com/status-im/status-mobile/issues/18613 )
2. Open watch only collectible

### Actual result:
'Send' button is shown for watch-only collectibles and can navigate to the 'send' flow
image



### Expected result:
Watch-only collectible page is shown as per design without 'send' button
image

https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=12077-156467&mode=design&t=Qzwvre6uKEljOmd6-0
### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 3: The select assets screen is shown when a watch-only token is going to be sent

### Steps to reproduce:
1. Add watch only an account with some assets that are not available on the regular accounts (example: `0x8d2413447ff297d30bdc475f6d5cb00254685aae`)
2. Long tap on the asset that is available only within the current watch-only account
3. Select an address to 'send to'
4. Check next page


### Actual result:
The 'Select assets' page is shown, even though the asset has already been selected in step 2.

https://github.com/status-im/status-mobile/assets/52490791/51afec6b-0175-4a6f-af67-71c0e62af1ff

### Expected result:
Potential solutions:
- The 'Select assets' page should not be shown since the asset has already been selected.
- probably if the asset is related only to a watch-only account, the drawer should not be shown. Let's wait for the design team's response https://discord.com/channels/624634427930312714/852533718790570015/1228346181097230539

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 4: The 'From' page is shown after sending the collectible from the account overview page

### Steps:
1. Restore the account with collectibles
2. Long tap collectible on the account overview page
3. Tap 'send'

### Actual result:
The 'from' screen is shown

https://github.com/status-im/status-mobile/assets/52490791/4bf7b0a8-8b8a-4fc2-8f2e-d99c253b7fa4

### Expected result:
the 'from' screen is missed for collectibles sending flow


### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by VolodLytvynenko** [comment]
## PR_ISSUE 5: The routes are not found when attempting to send a collectible using the 'send' button in the opened collectible

### Steps:
1. Restore the account with collectibles
2. Long tap collectible
3. Tap 'send'

### Actual result:
The routes are not found

https://github.com/status-im/status-mobile/assets/52490791/d90ccee5-ae90-4f86-879f-91a1e793e0a3

### Expected result:
The routes are found

### Logs:
[Status-debug-logs (8).zip](https://github.com/status-im/status-mobile/files/14960326/Status-debug-logs.8.zip)

**Comment by ulisesmac** [comment]
Hey @VolodLytvynenko 
sorry for the late response.

### Issue 1

This is due to the relogin issue, so I'd say it's out of scope of this one

### Issue 2 and 3

This PR is not focused on fixing UI things, so they need to be addressed but preferably in a separate one.

### Issue 4
This issue is also related to the recovery and relogin one, so I think we should address it separately.

### Issue 5

We don't know the exact reason for routes not being found, but it's out of scope of this PR.


So, this PR is about connecting some flows, but they may still be buggy for various reasons, we'll address them separately.

**Comment by VolodLytvynenko** [comment]
Hi @ulisesmac thank you for clarification. PR is ready to be merged

**Comment by VolodLytvynenko** [comment]
@ulisesmac please let me know if any additional tests are needed after resolving the conflicts in this PR. I can quickly check everything

**Comment by ulisesmac** [comment]
@VolodLytvynenko I solved the conflicts  and merged the PR, I'll open issues for the points you pointed out and I'll be opening PRs to address them. Thank you!

**Comment by ulisesmac** [comment]
> ## PR_ISSUE 3: The select assets screen is shown when a watch-only token is going to be sent
> ### Steps to reproduce:
> 
>     1. Add watch only an account with some assets that are not available on the regular accounts (example: `0x8d2413447ff297d30bdc475f6d5cb00254685aae`)
> 
>     2. Long tap on the asset that is available only within the current watch-only account
> 
>     3. Select an address to 'send to'
> 
>     4. Check next page
> 
> 
> ### Actual result:
> 
> The 'Select assets' page is shown, even though the asset has already been selected in step 2.
> select_as_page.mp4
> ### Expected result:
> 
> Potential solutions:
> 
>     * The 'Select assets' page should not be shown since the asset has already been selected.
> 
>     * probably if the asset is related only to a watch-only account, the drawer should not be shown. Let's wait for the design team's response https://discord.com/channels/624634427930312714/852533718790570015/1228346181097230539
> 
> 
> ### OS:
> 
> IOS, Android
> ### Devices:
> 
>     * Pixel 7a, Android 13
> 
>     * iPhone 11 Pro Max, IOS 17


An update @VolodLytvynenko 

This is being addressed by @mmilad75  in:
- https://github.com/status-im/status-mobile/pull/20050

**Comment by ulisesmac** [comment]
@VolodLytvynenko Another update on the issues, I've tested them on latest develop:

> PR_ISSUE 1: The assets can't be sent on 'send to' page when an asset is selected on account overview page

No longer reproducible.

> PR_ISSUE 2: 'Send' button is shown for watch-only collectibles and can navigate to the 'send' flow

No longer reproducible.

> PR_ISSUE 3: The select assets screen is shown when a watch-only token is going to be sent

As said above @mmilad75  is solving it.

> PR_ISSUE 4: The 'From' page is shown after sending the collectible from the account overview page

No longer reproducible.

> PR_ISSUE 5: The routes are not found when attempting to send a collectible using the 'send' button in the opened collectible

No longer reproducible. Maybe it was happening because the collectible didn't support sending, you didn't have enough eth or it was a side issue and now it's solved.


So there's nothing else pending from this PR, thank you for taking the time to test!

**Comment by VolodLytvynenko** [comment]
> The select assets screen is shown when token is going to be sent which available on both watch-only and regular account #19745

Hi @ulisesmac thank you for the update. I confirm this. Issues you mentioned are fixed
